src/main/java/org/wso2/siddhi/plugins/idea/completion/definitionelements/DefinitionCompletionContributor.java

```diff
@@ -15,8 +15,224 @@
+import org.wso2.siddhi.plugins.idea.psi.WindowDefinitionNode;
+
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addAggregateKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addAggregationTimeTypesKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addAtKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addByKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addDefineTypesAsLookups;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addEveryKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addFromKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addGroupKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addLanguageTypesKeywords;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addOutputEventTypeKeywords;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addReturnKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addSelectKeyword;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addValueTypesAsLookups;
+import static org.wso2.siddhi.plugins.idea.completion.SiddhiCompletionUtils.addWindowProcessorTypesAsLookups;
+import static org.wso2.siddhi.plugins.idea.completion.util.KeywordCompletionUtils.getPreviousVisibleSiblingSkippingComments;
+
 /**
  * Provides code completions for definitions.
  */
 public class DefinitionCompletionContributor {
+
+    public static void definitionCompletion(@NotNull CompletionResultSet result, PsiElement element,
+                                           PsiElement prevVisibleSibling,
+                                           IElementType prevVisibleSiblingElementType,
+                                           PsiElement prevPrevVisibleSibling) {
+        // suggestions after define keyword
+        if (prevVisibleSiblingElementType == SiddhiTypes.DEFINE) {
+            addDefineTypesAsLookups(result);
+            return;
+        }
+        // suggestions after an attribute type if it is in a definition element other than an aggregation definition
+        // if the aggregation is also included it will provide attribute types after the attributes in 'select'
+        // clause inside the aggregation definition
+        if (PsiTreeUtil.getParentOfType(prevVisibleSibling, AttributeNameNode.class) != null
+                && PsiTreeUtil.getParentOfType(prevVisibleSibling, DefinitionElementNode.class) != null
+                && PsiTreeUtil.getParentOfType(prevVisibleSibling, AggregationDefinitionNode.class) == null) {
+            addValueTypesAsLookups(result);
+            return;
+        }
+        //TODO: on that class handle attribute type suggestions
```

Comment:
Is this comment redundant or this still needs to be done?

Reply:
This is still needed to be done